src/client/extensionActivation.ts

```diff
@@ -83,25 +83,12 @@ async function activateLegacy(
     const fs = serviceManager.get<IFileSystem>(IFileSystem);
     await setExtensionInstallTelemetryProperties(fs);
 
-    const applicationEnv = serviceManager.get<IApplicationEnvironment>(IApplicationEnvironment);
-    const enableProposedApi = applicationEnv.packageJson.enableProposedApi;
-    serviceManager.addSingletonInstance<boolean>(UseProposedApi, enableProposedApi);
-    // Feature specific registrations.
-    variableRegisterTypes(serviceManager);
-    installerRegisterTypes(serviceManager);
-    commonRegisterTerminalTypes(serviceManager);
-
-    const configuration = serviceManager.get<IConfigurationService>(IConfigurationService);
-    // We should start logging using the log level as soon as possible, so set it as soon as we can access the level.
-    // `IConfigurationService` may depend any of the registered types, so doing it after all registrations are finished.
-    // XXX Move this *after* abExperiments is activated?
-    setLoggingLevel(configuration.getSettings().logging.level);
-
     // Load the two data science experiments that we need to register types
     // Await here to keep the register method sync
     const experimentService = serviceContainer.get<IExperimentService>(IExperimentService);
     experimentService.logExperiments();
-    let useVSCodeNotebookAPI = await experimentService.inExperiment(Experiments.NativeNotebook);
+    let useVSCodeNotebookAPI =
+        env.appName.includes('Insider') || (await experimentService.inExperiment(Experiments.NativeNotebook));
     let inCustomEditorApiExperiment = await experimentService.inExperiment(Experiments.CustomEditor);
```

Comment:
I think inCustomEditorApiExperiment should be false if nativeEditor is true.

Reply:
Maybe in the experiment service level? So nothing can have both of them turned on at the same time.

Reply:
This is done further below in that code.